lib/tasks_screen/bloc/task_bloc.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:async';
+import 'package:bloc/bloc.dart';
+import 'package:flutter/foundation.dart';
+import 'package:lachies_life_planner/tasks_screen/bloc/task_event.dart';
+import 'package:lachies_life_planner/tasks_screen/bloc/task_state.dart';
+import 'package:lachies_life_planner/tasks_screen/models/task.dart';
+import 'package:lachies_life_planner/tasks_screen/models/task_repository.dart';
+
+class TasksBloc extends Bloc<TasksEvent, TasksState> {
+  final TasksRepository tasksRepository;
+  StreamSubscription _tasksSubscription;
+
+  TasksBloc({@required this.tasksRepository}) : super(TasksLoadingState());
```

Comment:
Looks like we have duplicate logic.  We do `super(TasksLoadingState())` in the constructor, and then `yield TasksLoadingState()` inside `_mapLoadTasksToState()`

Reply:
If we remove the `super` call, there is an error.  And there seem to be no behaviour changes when removing that yield line, so I will just do that.

![image](https://user-images.githubusercontent.com/47808687/114291528-782a2a80-9acb-11eb-9874-538443c236ae.png)